File: python/tests/length_of_last_word_tests.py
import unittest

from python.tasks.length_of_last_word import Solution


class LengthOfLastWordTests(unittest.TestCase):

    def test_sum(self):
        test_data = (
            [
                (5, "Hello World"),
                (4, "   fly me   to   the moon  "),
                (6, "luffy is still joyboy"),
            ],
        )

        solution = Solution()

        for data in test_data:
            for expected_result, s in data:
                actual_result = solution.lengthOfLastWord(s)
                self.assertEqual(expected_result, actual_result)


if __name__ == "__main__":
    unittest.main()

File: csharp/LeetCodePractice.Tests/LengthOfLastWordTests.cs
namespace LeetCodePractice.Tests;

public class LengthOfLastWordTests
{
    [Theory]
    [MemberData(nameof(Data))]
    public void Tests(int result, string s)
    {
        var solution = new Console.LeetCodeTasks.LengthOfLastWord.Solution();

        var actualResult = solution.LengthOfLastWord(s);

        Assert.Equal(result, actualResult);
    }

    public static IEnumerable<object[]> Data =>
    [
        [5, "Hello World"],
        [4, "   fly me   to   the moon  "],
        [6, "luffy is still joyboy"],
        [0, ""],
    ];
}
